### PR TITLE
feat(protocol-designer): batch edit form skeleton

### LINKED ISSUE #7129: PD: Show "no shared settings" for multiple step types selected
### User Story
As a user, when I have a selection that isn't solely made up of Transfers, I'd like to know why there are no settings to edit.  

Please note: This story does not include a Mix batch edit form as an exception to the rule. The Mix batch edit form is marked as a fast follow. 

### Design
https://www.figma.com/file/6p1uxeeioDikcfSwLCfxi4/Batch-Editing?node-id=705%3A16532

### Acceptance Critieria
(NOTE: after functionality in #7350, this ticket should just need styling to match the design)

- [ ] When a set of steps are selected that isn't only Transfers, I see a different state than the batch edit form

**Test Steps**

1. Create a protocol with at least 2 transfer steps, 2 mix steps, 1 pause, 1 magnet, 1 thermocycler, 1 temp step
2. Select multiple transfer steps to enter batch edit mode. Batch edit transfer form should appear with all advanced settings editable
3. Add a mix step. Transfer form should change to new state, indicating to user that no advanced settings are shared between selected steps
4. Deselect the mix step. Batch edit form should be available again for transfer steps
5. Add a pause step. User should see "no advanced settings are shared" state.
6. Repeat steps 2-5 with remaining step types. Expected results: if only transfer steps are selected, batch edit form is displayed and editable, if any other step type is selected the batch edit form changes state (no advanced settings shared).
7. Deselect all transfer steps. Select a few random assortments of non-transfer, non-mix steps. Form should display no advanced settings shared

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview

Sets up batch edit form display (and hiding the single-step step edit form when in batch edit mode)

Addresses #7129 and addresses #7149 but doesn't close them (just placeholder text)

# Changelog

- New `FormManager`
- Refactor StepSelectionBanner to be dumber and take more props from FormManager
- Skeleton for batch edit form and batch edit > transfer form

## Update

After we did the pair review on this, I realized that https://www.figma.com/file/6p1uxeeioDikcfSwLCfxi4/Batch-Editing?node-id=705%3A16532 "No advanced settings shared between selected steps" isn't really a form, but is the whole panel (and replaces the deck map). I added the placeholder text and a TODO punted to #7129 
 
# Review requests

- [ ] Should show placeholder text fulfulling reqs of #7129 (but not styled)
- [ ] Should not cause any changes to PD when batch edit mode FF is off!
- [ ] Appropriate selectors?
- [ ] Cool committing these TODOs and placeholders? Do all the TODOs have tickets?

# Risk assessment

Medium, behind FF but could theoretically mess up single edit form's display

### CODE REVIEW

File: protocol-designer/src/components/BatchEditForm/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  fieldValues: MultiselectFieldValues | null,
+  handleChangeFormInput: (name: string, value: mixed) => void,
+|}
+
+type BatchEditMoveLiquidProps = {|
+  propsForFields: FieldPropsByName,
+  handleClose: () => mixed,
+  handleSave: () => mixed,
+|}
+export const BatchEditMoveLiquid = (
```
Comment (shlokamin) [inline protocol-designer/src/components/BatchEditForm/index.js:22]: nit pick: does this need to be exported?

File: protocol-designer/src/components/BatchEditForm/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  props: BatchEditMoveLiquidProps
+): React.Node => {
+  const { propsForFields, handleClose, handleSave } = props
+  return (
+    // TOOD IMMEDIATELY copied from SourceDestFields. Refactor to be DRY
```
Comment (shlokamin) [inline protocol-designer/src/components/BatchEditForm/index.js:27]: `TOOD` 😂 . Did you mean to do this as part of this PR or should we do this in a follow up?

File: protocol-designer/src/ui/steps/selectors.js [hunk 3]
```diff
@@ -388,3 +390,32 @@ export const getMultiSelectDisabledFields: Selector<DisabledFields | null> = cre
...
   }
 )
+
+export const getCountPerStepType: Selector<CountPerStepType> = createSelector(
+  getMultiSelectItemIds,
+  stepFormSelectors.getSavedStepForms,
+  (stepIds, allSteps) => {
+    if (stepIds === null) return {}
+
+    const steps = stepIds.map(id => allSteps[id])
```
Comment (shlokamin) [inline protocol-designer/src/ui/steps/selectors.js:400]: seems like another good reason to merge `ui/steps` and `step-forms`, they seem to be very very similar concerns

**Review by shlokamin — APPROVED** [review]
There are a few `TODO IMMEDIATELY' notes which we should be sure to address, but I'd like to merge this. Batch edit work is behind a FF and I'd rather this branch not get stale. Thanks @IanLondon!

**Review by Kadee80 — APPROVED** [review]
🍴 🥞

File: protocol-designer/src/components/BatchEditForm/index.js (hunk 2 above)
Comment (IanLondon) [reply protocol-designer/src/components/BatchEditForm/index.js:27]: should be part of #7128 work when we make this an actual form, I'll add to AC

File: protocol-designer/src/components/BatchEditForm/index.js (hunk 1 above)
Comment (IanLondon) [reply protocol-designer/src/components/BatchEditForm/index.js:22]: I figured once we make it a real form we'll want to import it to test it?

**Pull request merged**